tools/leader_transfer_command_generator/leader_transfer_command_generator.py

```diff
@@ -0,0 +1,168 @@
+#    -p <partitions> \
+#    -c <cores>
+#
+# Example:
+#  ./leader_transfer_command_generator.py -f /tmp/testdata.txt --source 1 --targets 2,3 \
+#     -t test,test2,test3 \
+#     -p 0,1,2,3,4,5 \
+#     -c 2
+#  transfer leadership for topics ['test', 'test2', 'test3'] / partitions [0, 1, 2, 3, 4, 5] on core [2] from 1 to [2, 3]
+#  curl -XPOST http://localhost:9644/v1/partitions/test/1/transfer_leadership?target=2
+#  curl -XPOST http://localhost:9644/v1/partitions/test3/4/transfer_leadership?target=3
+#
+
+
+def main():
+    import argparse
+    import itertools
+    import json
+
+    def generate_options():
+        parser = argparse.ArgumentParser(
+            description='Redpanda Granular Leader Transfer Command Generator')
+        parser.add_argument('-f',
+                            '--file',
+                            required=True,
+                            type=str,
+                            help='path to partitions file')
+        parser.add_argument('--source',
+                            required=True,
+                            type=str,
+                            help='a single source node id')
+        parser.add_argument('--targets',
+                            required=True,
+                            type=str,
+                            help='comma separated target node ids')
+        parser.add_argument('-t',
+                            type=str,
+                            help='comma separated user topics (Default: all)')
+        parser.add_argument('-p',
+                            type=str,
+                            help='comma separated partitions (Default: all)')
+        parser.add_argument(
+            '-c',
+            type=str,
+            help=
+            'comma separated core ids to where current leader is stuck (Default: all)'
+        )
+        return parser
+
+    parser = generate_options()
+    options, _ = parser.parse_known_args()
+    topics, partitions, cores = None, None, None
+    source, targets = None, None
+    exclude_list = ['controller', 'id_allocator', 'tx']
+
+    with open(options.file) as f:
+        all_partitions = json.load(f)
+
+    for i, p in enumerate(all_partitions):
+        if p['topic'] in exclude_list:
+            all_partitions.pop(i)
+
+    all_partitions.sort(key=lambda x: x['topic'])
+
+    # Preparation
+    if options.t is not None:
+        topics = options.t.split(',')
+
+    if options.p is not None:
+        if options.t is None:
+            print("partition id has to be specified along with topic name")
+            exit(1)
+        else:
+            partitions = [int(i) for i in options.p.split(',')]
+
+    if options.c is not None:
+        cores = [int(i) for i in options.c.split(',')]
+
+    source = int(options.source)
+    targets = [int(i) for i in options.targets.split(',')]
+
+    # Parsing all the partitions
+
+    # No option is specified
+    if topics is None and partitions is None and cores is None:
+        print(
```

Comment:
For these explanatory `print()`s, can we maybe send them to stderr to make redirecting the output a bit easier? Otherwise you'd need to strip this line every time before doing anything with the output.

Reply:
Indeed, great idea. Thanks :)